src/components/Layout/Header/NotificationCenter/components/SwapActionCard.tsx

```diff
@@ -31,41 +35,38 @@ const hoverProps = {
   cursor: 'pointer',
 }
 
-type NotificationCardProps = {
-  type: NotificationType
-  assetId: AssetId
-  secondaryAssetId?: AssetId
-  status: NotificationStatus
-  date: number
-  title: string
+type SwapActionCardProps = {
   isCollapsable?: boolean
   defaultIsOpen?: boolean
-} & PropsWithChildren
+} & Action &
+  PropsWithChildren
 
-export const NotificationCard = ({
-  type,
-  assetId,
-  secondaryAssetId,
-  status,
-  date,
-  title,
+export const SwapActionCard = ({
   children,
   isCollapsable = true,
   defaultIsOpen = false,
-}: NotificationCardProps) => {
+  ...action
+}: SwapActionCardProps) => {
   const { isOpen, onToggle } = useDisclosure({ defaultIsOpen })
+  const swapsById = useAppSelector(swapSlice.selectors.selectSwapsById)
 
   const formattedDate = useMemo(() => {
     const now = dayjs()
-    const notificationDate = dayjs.unix(date)
+    const notificationDate = dayjs(action.createdAt)
     const sevenDaysAgo = now.subtract(7, 'day')
 
     if (notificationDate.isAfter(sevenDaysAgo)) {
       return notificationDate.fromNow()
     } else {
       return notificationDate.toDate().toLocaleString()
     }
-  }, [date])
+  }, [action.createdAt])
+
+  const swap = useMemo(() => {
+    if (isSwapAction(action)) {
```

Comment:
q: shouldn't `action` inherently already be narrowed down by virtue of this being a `<SwapActionCard />` ?

Reply:
Makes a lot of sense, changed in [b1ba80f](https://github.com/shapeshift/web/pull/9625/commits/b1ba80f90da2bc1ee837997769ab467d9ad2048b)